overlord/devicestate/handlers.go

```diff
@@ -939,8 +939,12 @@ func currentGadgetInfo(snapst *snapstate.SnapState) (*gadget.GadgetData, error)
 		// no current yet
 		return nil, nil
 	}
-	const onClassic = false
-	gi, err := gadget.ReadInfo(currentInfo.MountDir(), onClassic)
+
+	constraints := &gadget.ModelConstraints{
+		Classic:    false,
+		SystemSeed: release.HasSystemSeed,
```

Comment:
I wonder if we should simply pass "nil" here to ReadInfo() to indicate that it should do a self-consistency check at this point. This would mean we can get rid of release.HasSystemSeed feels a bit out-of-place to me here (also at this point nothing will write partitions to disk anymore). If I simply pass "nil" here no test breaks wihch indicates that its either fine to do that or that we need an extra test if it is not :)

Reply:
this is premature, what I recommended originally is to split this first in adding ModelConstraints with classic only and then do the core20 changes. We will need to get the model in here to do the right thing in the end.

Reply:
Right. I'll remove this setting and wait until we get the rest of the infrastructure in place.